code/modules/tgui/tgui.dm

```diff
@@ -203,6 +203,13 @@
 		html = replacetextEx(SStgui.basehtml, "{}", get_json(initial_data))
 	else
 		html = SStgui.basehtml
+
+	//Allow the src object to override the html if needed
+	html = src_object.ui_base_html(html)
+	//Strip out any remaining custom tags that are used in ui_base_html
+	html = replacetext(html, "(customheadhtml)", "")

```

Comment:
Should probably be done before inline mode data gets inserted.

Reply:
I don't understand what you mean

Reply:
Two lines above, admins with debug get initial data pasted into html (i guess so you can copy whole thing and debug externally), should probably be before that in case data contains (customheadhtml) somewhere.

Reply:
okay that makes sense